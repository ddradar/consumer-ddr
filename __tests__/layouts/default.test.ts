import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils'
import Buefy from 'buefy'

import DefaultLayout from '~/layouts/default.vue'

type MenuItem = {
  title: string
  subMenu: { name: string; href: string }[]
}

const localVue = createLocalVue()
localVue.use(Buefy)

describe('layouts/default.vue', () => {
  test('renders correctly', () => {
    // Arrange - Act
    const menuList: MenuItem[] = [
      {
        title: 'Play Station (JP)',
        subMenu: [
          { name: 'Dance Dance Revolution', href: '/series/1st-jp/' },
          {
            name: 'Dance Dance Revolution 2nd ReMIX',
            href: '/series/2nd-remix-jp/'
          }
        ]
      },
      {
        title: 'GAME BOY COLOR',
        subMenu: [{ name: 'Dance Dance Revolution GB', href: '/series/gb-1/' }]
      },
      {
        title: 'Play Station 2 (US)',
        subMenu: [{ name: 'Dance Dance Revolution X2', href: '/series/x2-us/' }]
      }
    ]
    const wrapper = mount(DefaultLayout, {
      localVue,
      stubs: { NuxtLink: RouterLinkStub, Nuxt: true },
      data: () => ({ menuList })
    })

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })
})
