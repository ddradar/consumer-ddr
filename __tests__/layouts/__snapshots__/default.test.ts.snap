// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts/default.vue renders correctly 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar is-primary"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item has-text-weight-bold"
      >
        <img
          alt="Logo"
          class="images"
          src="~assets/img/logo.svg"
        />
        
        Consumer DDR
      
      </a>
      <a
        aria-label="menu"
        class="navbar-burger burger"
        role="button"
      >
        <span
          aria-hidden="true"
        />
         
        <span
          aria-hidden="true"
        />
         
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
        >
          Songs
        </a>
         
        <div
          class="navbar-item has-dropdown"
        >
          <a
            aria-haspopup="true"
            class="navbar-link"
            href="#"
            role="menuitem"
          >
            Play Station (JP)
          </a>
           
          <div
            class="navbar-dropdown"
          >
            <a
              class="navbar-item"
            >
              
          Dance Dance Revolution
        
            </a>
            <a
              class="navbar-item"
            >
              
          Dance Dance Revolution 2nd ReMIX
        
            </a>
          </div>
        </div>
        <div
          class="navbar-item has-dropdown"
        >
          <a
            aria-haspopup="true"
            class="navbar-link"
            href="#"
            role="menuitem"
          >
            GAME BOY COLOR
          </a>
           
          <div
            class="navbar-dropdown"
          >
            <a
              class="navbar-item"
            >
              
          Dance Dance Revolution GB
        
            </a>
          </div>
        </div>
        <div
          class="navbar-item has-dropdown"
        >
          <a
            aria-haspopup="true"
            class="navbar-link"
            href="#"
            role="menuitem"
          >
            Play Station 2 (US)
          </a>
           
          <div
            class="navbar-dropdown"
          >
            <a
              class="navbar-item"
            >
              
          Dance Dance Revolution X2
        
            </a>
          </div>
        </div>
      </div>
      <div
        class="navbar-end"
      />
    </div>
  </nav>
   
  <section
    class="main-content"
  >
    <nuxt-stub />
  </section>
</div>
`;
